### PR TITLE
[Merl 797] Feat: fromThemeJson helper

### PR DESCRIPTION
**Pull request by theodesp**
## Tasks

- [ ] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.

## Description

This PR exposes the following helper in the @faustwp/blocks package:

**fromThemeJson**: The helper function that accepts a theme.json parameter and maps this type into the `BlocksTheme` schema object.
## Testing

* Take a look at the test cases and check how it works
* Use this function as an input to `WordpressBlocksProvider`. It should work as expected:
```
import themeJson from "theme.json"; // import common theme.json here
import {fromThemeJson} from "@faustwp/blocks";

<WordpressBlocksProvider
      config={{
      theme={fromThemeJson(themeJson}}
...
```

### CODE REVIEW

File: packages/blocks/src/utils/from-theme-json/fromThemeJson.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Parses the passed theme.json object and returns a valid Block Theme schema.
+ * @param theme The theme.json object
+ */
+export function fromThemeJson(theme: Record<string, any>): BlocksTheme {
+  const result: BlocksTheme = reduce(
+    theme,
+    (acc, value, key) => {
+      if (key === 'settings') {
```
Comment (theodesp) [inline packages/blocks/src/utils/from-theme-json/fromThemeJson.ts:14]: Just map over `settings` and `styles` from theme.json into the schema.

File: packages/blocks/src/utils/from-theme-json/getStylesOptions.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * Takes an object with an optional `palette` property and transforms it into the ThemePropertiesColor object type.
+ * @param {StylesProperties} styles - An object of type StylesProperties.
+ * @returns {ThemePropertiesColor} An object with an optional `palette` property of type `Palette`.
+ */
+export function getStylesOptions(
+  styles?: StylesProperties,
+): ThemePropertiesStyles {
+  if (!isEmpty(styles)) {
+    return {
+      ...styles,
```
Comment (theodesp) [inline packages/blocks/src/utils/from-theme-json/getStylesOptions.ts:27]: Not much of a change here. We just map styles (including blocks overrided) as it is.

File: packages/blocks/tests/utils/from-theme-json/fromThemeJson.test.ts [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+    const expectedResult = {};
+    expect(fromThemeJson(themeJson)).toEqual(expectedResult);
+  });
+
+  it('returns a BlocksTheme object when passed a theme.json object with settings and styles', () => {
+    const themeJson = theme;
+
+    expect(fromThemeJson(themeJson)).toMatchInlineSnapshot(`
+      Object {
+        "fontFamilies": Object {
```
Comment (theodesp) [inline packages/blocks/tests/utils/from-theme-json/fromThemeJson.test.ts:16]: Our schema is a more compact version of the theme.json with only the relevant parts. We might improve this on a use-case basis.

**Review by blakewilson — APPROVED** [review]
Nice @theodesp! Love the tests.

**Pull request merged**